modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/runtime/compatibility/DefaultForwardCompatibilityHelper.java

```diff
@@ -6,7 +6,41 @@
  */
 package org.mule.runtime.module.extension.api.runtime.compatibility;
 
-import org.mule.sdk.api.utils.ForwardCompatibilityHelper;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.internal.profiling.InternalProfilingService;
+import org.mule.runtime.extension.api.runtime.parameter.CorrelationInfo;
+import org.mule.runtime.module.extension.internal.runtime.parameter.ImmutableCorrelationInfo;
+import org.mule.runtime.module.extension.internal.runtime.resolver.DistributedTraceContextManagerResolver;
+import org.mule.runtime.module.extension.internal.runtime.source.trace.DefaultDistributedSourceTraceContext;
+import org.mule.sdk.api.runtime.source.DistributedTraceContextManager;
+import org.mule.sdk.compatibility.api.utils.ForwardCompatibilityHelper;
+
+import javax.inject.Inject;
 
 public class DefaultForwardCompatibilityHelper implements ForwardCompatibilityHelper {
+
+  @Inject
+  private InternalProfilingService profilingService;
```

Comment:
DefaultForwardCompatibilityHelper is loaded as a service so it is not managed by the registry and nothing will be injected. For this to be injected, you can inject the dependencies programmatically for example muleContext.getInjector().inject(helper) or you can define define the helper as a bean (I don't know if this is what is intended)
Possibly as this is a helper you can pass the coreEventTracer to the method as a parameter and the method can be static (so nothing is injected through @Inject). The places where you invoke those methods should be in a place that is managed by the mule runtime registry so that you can pass the profiling server or the coreEvenTracer that you retrieved from that injected profiling server.